src/Employer/Employer.Web/Orchestrators/EditVacancyDatesOrchestrator.cs

```diff
@@ -40,15 +39,18 @@ public async Task<Vacancy> GetVacancyAsync(VacancyRouteModel vrm)
 
             Utility.CheckAuthorisedAccess(vacancy, vrm.EmployerAccountId);
 
+            if (vacancy.CanExtendStartAndClosingDates == false)
+                throw new InvalidStateException(string.Format(ViewModels.ErrorMessages.VacancyNotAvailableForEditing, vacancy.Title));
+
             return vacancy;
         }
 
-        public Task<OrchestratorResponse<EditVacancyDatesViewModel>> GetEditVacancyDatesViewModel(VacancyRouteModel vrm, string proposedClosingDate, string proposedStartDate)
+        public async Task<OrchestratorResponse<EditVacancyDatesViewModel>> GetEditVacancyDatesViewModelAsync(VacancyRouteModel vrm, DateTime? proposedClosingDate, DateTime? proposedStartDate)
         {
             var vacancyTask = GetVacancyAsync(vrm);
             var programmesTask = _client.GetActiveApprenticeshipProgrammesAsync();
 
-            Task.WaitAll(vacancyTask, programmesTask);
+            await Task.WhenAll(vacancyTask, programmesTask);
```

Comment:
👍 - Is this where the aggregate exception was being thrown from?

Reply:
Yep. There are a few other WaitAlls in the code so I've changed this one. We may want to revisit the other ones at some point.